drivers/net/adin1110/adin1110.c

```diff
@@ -0,0 +1,820 @@
+
+#define ADIN1110_CRC_POLYNOMIAL	0x7
+
+NO_OS_DECLARE_CRC8_TABLE(_crc_table);
+
+struct _adin1110_priv {
+	uint32_t phy_id;
+	uint32_t num_ports;
+};
+
+static struct _adin1110_priv driver_data[2] = {
+	[ADIN1110] = {
+		.phy_id= ADIN1110_PHY_ID,
+		.num_ports = 1,
+	},
+	[ADIN2111] = {
+		.phy_id = ADIN2111_PHY_ID,
+		.num_ports = 2,
+	},
+};
+
+/**
+ * @brief Write a register's value
+ * @param desc - the device descriptor
+ * @param addr - register's address
+ * @param data - register's value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int adin1110_reg_write(struct adin1110_desc *desc, uint16_t addr, uint32_t data)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->tx_buff,
+		.bytes_number = ADIN1110_WR_FRAME_SIZE,
+		.cs_change = 1,
+	};
+
+	/** The address is 13 bit wide */
+	addr &= ADIN1110_ADDR_MASK;
+	addr |= ADIN1110_CD_MASK | ADIN1110_RW_MASK;
+	no_os_put_unaligned_be16(addr, desc->tx_buff);
+	no_os_put_unaligned_be32(data, &desc->tx_buff[2]);
+
+	return no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+}
+
+/**
+ * @brief Read a register's value
+ * @param desc - the device descriptor
+ * @param addr - register's address
+ * @param data - register's value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int adin1110_reg_read(struct adin1110_desc *desc, uint16_t addr, uint32_t *data)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->tx_buff,
+		.rx_buff = desc->rx_buff,
+		.bytes_number = ADIN1110_RD_FRAME_SIZE,
+		.cs_change = 1,
+		.cs_change_delay = 5,
+	};
+	int ret;
+
+	no_os_put_unaligned_be16(addr, &desc->tx_buff[0]);
+	desc->tx_buff[0] |= NO_OS_BIT(7);
+	desc->tx_buff[2] = 0x00;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	*data = no_os_get_unaligned_be32(&desc->rx_buff[3]);
+
+	return 0;
+}
+
+/**
+ * @brief Update a register's value based on a mask
+ * @param desc - the device descriptor
+ * @param addr - register's address
+ * @param mask - the bits that may be modified
+ * @param data - register's value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int adin1110_reg_update(struct adin1110_desc *desc, uint16_t addr,
+			uint32_t mask, uint32_t data)
+{
+	uint32_t val;
+	int ret;
+
+	ret = adin1110_reg_read(desc, addr, &val);
+	if (ret)
+		return ret;
+
+	val &= ~mask;
+	val |= mask & data;
+
+	return adin1110_reg_write(desc, addr, val);
+}
+
+/**
+ * @brief Read a PHY register using clause 22
+ * @param desc - the device descriptor
+ * @param phy_id - the phy device's id
+ * @param reg - register's address
+ * @param data - register's value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int adin1110_mdio_read(struct adin1110_desc *desc, uint32_t phy_id,
+		       uint32_t reg, uint32_t *data)
+{
+	uint32_t mdio_val = 0;
+	uint32_t mdio_done = 0;
+	uint32_t val = 0;
+	int ret;
+
+	/* Use clause 22 for the MDIO register addressing */
+	val |= no_os_field_prep(ADIN1110_MDIO_ST, 0x1);
```

Comment:
i'd drop all these `val |=` and assign only once with `val =`. Check throughout the entire code for this occurrence.